src/applications/mhv-medical-records/actions/downloads.js

```diff
@@ -0,0 +1,57 @@
+import { generateCCD, downloadCCD } from '../api/MrApi';
+import { Actions } from '../util/actionTypes';
+import { addAlert } from './alerts';
+import * as Constants from '../util/constants';
+
+export const genAndDownloadCCD = (firstName, lastName) => async dispatch => {
+  dispatch({ type: Actions.Downloads.GENERATE_CCD });
+  try {
+    // GET LIST OF CCDs
+    const generate = await generateCCD();
+
+    // SUCCESSFUL CCD GENERATION
+    if (generate[0]?.status === 'COMPLETE') {
+      // getting the timestamp and filename for download
+      const timestamp = generate[0].dateGenerated;
+      const timestampDate = new Date(timestamp);
+      const fileName = `VA-Continuity-of-care-document-${
+        firstName !== null ? `${firstName}-` : ''
```

Comment:
Can `firstName` ever be undefined? Rather than checking null, can you just check all truthy values? (e.g. `{firstname ? ...}`